File: 20221227_Tuesday/20221227.rb
# Leetcode: 2279. Maximum Bags With Full Capacity of Rocks.
# https://leetcode.com/problems/maximum-bags-with-full-capacity-of-rocks/
# = = = = = = = = = = = = = =
# Accepted.
# Thanks God, Jesus Christ!
# = = = = = = = = = = = = = =
# Runtime: 278 ms, faster than 100.00% of Ruby online submissions for Maximum Bags With Full Capacity of Rocks.
# Memory Usage: 224.6 MB, less than 100.00% of Ruby online submissions for Maximum Bags With Full Capacity of Rocks.
# 2022.12.27 Daily Challenge.
# @param {Integer[]} capacity
# @param {Integer[]} rocks
# @param {Integer} additional_rocks
# @return {Integer}
def maximum_bags(capacity, rocks, additional_rocks)
  delta = capacity.zip(rocks).map{|a,b| a-b}.sort
  counter = 0
  delta.each do |value|
    if value <= additional_rocks
      additional_rocks -= value
      value = 0
    end
    counter += 1 if 0 == value
  end
  return counter
end

File: 20221227_Tuesday/20221227.go
// Leetcode: 2279. Maximum Bags With Full Capacity of Rocks.
// https://leetcode.com/problems/maximum-bags-with-full-capacity-of-rocks/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 155 ms, faster than 100.00% of Go online submissions for Maximum Bags With Full Capacity of Rocks.
// Memory Usage: 9.2 MB, less than 7.14% of Go online submissions for Maximum Bags With Full Capacity of Rocks.
// 2022.12.27 Daily Challenge.
import "sort"
func maximumBags(capacity []int, rocks []int, additionalRocks int) int {
  delta := make([]int,len(rocks))
  for i,_ := range delta {
    delta[i] = capacity[i] - rocks[i]
  }
  sort.Ints(delta)
  counter := 0
  for _, v := range delta{
    if v <= additionalRocks {
      additionalRocks -= v
      v = 0
    }
    if 0 == v { counter += 1}
  }
  return counter
}
